src/main/java/org/galatea/starter/domain/HistoricalPricesCache.java

```diff
@@ -0,0 +1,117 @@
+package org.galatea.starter.domain;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class HistoricalPricesCache {
+    private String cacheFilePath;
+
+    private Map<String, Map<String, List<IexHistoricalPrice>>> rangedCache;
+    private Map<String, Map<LocalDate, List<IexHistoricalPrice>>> datedCache;
+
+    public HistoricalPricesCache(String cacheFilePath) {
+        this.cacheFilePath = cacheFilePath;
+
+        log.info("Attempting to load historical prices cache from disk. Loading from {}", cacheFilePath);
+
+        try(FileInputStream fi = new FileInputStream(new File(cacheFilePath))) {
+            log.info("Cache file found. Initialising cache...");
+
+            try (ObjectInputStream oi = new ObjectInputStream(fi)) {
+                rangedCache = (Map<String, Map<String, List<IexHistoricalPrice>>>) oi.readObject();
+                datedCache = (Map<String, Map<LocalDate, List<IexHistoricalPrice>>>) oi.readObject();
+
+                log.info("Cache loaded. Loaded {} ranged symbols and {} dated symbols", rangedCache.size(), datedCache.size());
+            }
+            catch (ClassNotFoundException ex) {
+                log.warn("Exception raised when loading from cache:\n{}\nInitialising empty cache.", ex);
```

Comment:
When logging an exception, we generally wouldn't include a `{}` placeholder in the log. As long as the exception is the last arg in the list, it'll be appended to the end of the log message, which is where we usually want it.